### DISCUSSION TITLE
ID auto generated is not followed up sequence

### DISCUSSION THREAD

**Discussion by peterburgs** [prisma/prisma#19237] [Q&A]
I am working with Prisma version 4.10.0. The IDs which Prisma generates are not correct.
Here is my model:
```
model Pass {
  id                          Int       @id @default(autoincrement())
  createdAt                   DateTime? @default(now())
  updatedAt                   DateTime? @updatedAt
}
```
In the beginning, it worked well, but the day after, when I started the development server again and made a Creation action, the ID became `3002` instead of followed after the ID `2006`.

id  |createdAt              |updatedAt              
----+-----------------------+-----------------------
2004|2023-05-13 14:24:26.758|2023-05-14 10:03:51.510
2005|2023-05-13 14:27:29.565|2023-05-14 03:52:38.615
**2006**|2023-05-13 14:34:05.500|2023-05-13 14:36:39.144
**3002**|2023-05-14 10:03:58.188|2023-05-14 10:04:13.252
3003|2023-05-14 10:13:01.459|2023-05-14 10:13:01.549
3004|2023-05-14 10:13:12.843|2023-05-14 10:13:12.920
3005|2023-05-14 10:13:16.848|2023-05-14 10:13:16.924
3006|2023-05-14 10:13:23.409|2023-05-14 10:13:23.492
3007|2023-05-14 10:13:28.844|2023-05-14 10:13:28.914
3008|2023-05-14 10:13:36.271|2023-05-14 10:13:36.374
3009|2023-05-14 10:13:43.451|2023-05-14 10:13:43.534
3010|2023-05-14 10:13:49.438|2023-05-14 10:13:49.518
3011|2023-05-14 10:15:40.584|2023-05-14 10:15:40.772

Could anyone tell me this is a bug or feature?

**Comment by ludralph** [comment]
Hi @peterburgs 👋 

This is not a desired behavior for autoincremented ID. Did you by any chance perform other operations that could have impacted the ID  generated? You will have gaps each time you have failed inserts, each time you delete records then insert new ones. This could possibly explain why the sequence is not in order.

**Reply by peterburgs** [reply]
@ludralph Thanks for your response. 
I am sure that I did not make any other actions.

Day 1: Start the server, and create some records. Shut down the laptop.
At the end of Day 1, the latest ID is 2006.

Day 2: Start the laptop, start the server, and create some records.
The ID started from 3002 and counting.

**Reply by ludralph** [reply]
That's a strange behavior. Can you share the repository so we can reproduce this?

**Reply by peterburgs** [reply]
It is an Internal system of my company, cannot share. 

But you can reproduce using the model I posted above, nothing more.

**Reply by ludralph** [reply]
Which database provider are you using?

**Reply by peterburgs** [reply]
I'm using SQL Server 2017-latest

**Reply by kdev** [reply]
I am experiencing the same error with Postgres.
Looking into the sequence table I can see, that the sequence is increasing but there are no entries being written at all.
I am using `skipDuplicates: true` and a `createMany` insert.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
